pkg/server/download.go

```diff
@@ -355,3 +361,18 @@ func (s *server) updateDownloadSubscribers(total int64, current int64, name stri
 		s.subscribers.Delete(id)
 	}
 }
+
+// If the source and destination are on different file systems this will cause
+// an invalid cross-link device error using os.Rename. Instead, open the file
+// and then write it out
```

Comment:
is it easy to detect if the error is a cross-link error? is it worth trying the rename first and then doing the read/write style?